packages/@uppy/xhr-upload/src/index.ts

```diff
@@ -235,20 +208,13 @@ export default class XHRUpload<
             },
           })
 
-          if (!this.opts.validateStatus(res.status, res.responseText, res)) {
-            throw new NetworkError(res.statusText, res)
-          }
-
-          const body = this.opts.getResponseData(res.responseText, res)
-          const uploadURL = body?.[this.opts.responseUrlFieldName] as
-            | string
-            | undefined
+          const body = JSON.parse(res.responseText) as B
 
           for (const file of files) {
             this.uppy.emit('upload-success', file, {
               status: res.status,
               body,
-              uploadURL,
+              uploadURL: body.url as string | undefined,
```

Comment:
why is this needed if body is already type asseted `as B`? shouldn't we change type `Body` so that it instead has `url`?

Reply:
Because we do not know what `B` is, it's `Record<string, unknown>`. Maybe we should consider creating specific `B` generics per uploader, but that's outside of the scope of this PR.